### PR TITLE
Second blog post for RGSoC added

### PR DESCRIPTION
**Pull request by mittalshravika**
Hey, @Frijol, I have added the second blog post for Tessel. You may review the same.

It consists of:

* Our experience
* Challenges
* What we plan for future

### CODE REVIEW

File: Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Team 276linesofCode - Rails Girls Summer of Code 2017
+
+About two and a half months back, none of us had an idea of what this journey would be like. RGSoC had been the first Summer of Code or any other program as such that we were a part of, and our eagerness to grasp the most from the program had us going all throughout. Not to mention, we have gained so much more than we had even anticipated.
```
Comment (Frijol) [inline Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: "or any other program as such" – "as such" isn't quite right here, maybe "or any code mentorship program", to highlight what about this program made it new to you?

Comment (Frijol) [inline Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: I'm not sure I understand "had us going all throughout". Is there another way to say this?

File: Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+We both believe that Tessel was the best project that we could have worked on since it welcomes new contributors in a positive manner that motivates them to give back to the community as much as they can.
```
Comment (Frijol) [inline Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: <3

File: Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+Our Hardware modules were shipped by the Tessel team from the US and we spent some time exploring them. This was the most exciting part - Making circuits, seeing the LEDs shimmer, watching the Accelerometer come back to life, and many many more.
+
+After exploring the different repositories in Tessel, we started exploring issues which were Contribution starters. A lot of them were listed in the Tessel blog called "[This Week in Tessel](https://tessel.io/blog/157835574022/this-week-in-tessel-fun-with-open-source)" and we started completing them one at a time.
```
Comment (Frijol) [inline Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: can you explain "contribution-starter" for the readers? This is an issue label unique to our project. And, this would be a great place to link to the label so others can see what good starter issues might be!

File: Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+We made important tutorial modules for Tessel for their documentation page. This included Fritzing diagrams of the circuits, describing the functionality in layman terms, and writing code which would be easily understood, with the help of the comments. Along with this we started working on "[reach](https://github.com/tessel/reach-wg)" during the last few weeks of the program.
+
+Apart from all the technical work, we learnt about Product Design and how exactly does a product make it's way to the Open Source domain. We made a draft of something called a Product Requirement Document (PRD) for our proposal of HAP (Humanoid Arm Project) that we had made during our application process.
```
Comment (Frijol) [inline Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: "it's" should be "its"

Comment (Frijol) [inline Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: "learnt" should be "learned"

File: Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+* Tessel was all based on JavaScript. We were both new to this language and hence faced certain challenges while comprehending the code which was already there in Tessel. But, our mentor Kelsey along with one of our coaches, Divam, helped us in figuring out the keywords used in the code snippets with the help of which we were able to make some tutorial for a few functionalities in Tessel in JavaScript.
+
+## A shoutout to the amazing Tessel Community!!
+
+Our blog couldn't be complete had we not thanked the awesome Tessel Community that made our descent into Open Source, a fun and a smooth ride. Every call with Kelsey would begin with her asking us whether we have been achieving whatever we planned for. She even asked if the tasks for the week were a lot or were they too less. But she always ensured that we had the right amount of work on our plate for that week.
```
Comment (Frijol) [inline Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: "too less" should be "too little"

File: Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+Our weekly calls with Kelsey, our mentor, were the times where we had learnt the most. The timely response of the community members on any of our doubts were commendable. Any PR review would come in a day or a two and the reviews gave us a different perspective on the issues. Nick and Kelsey, both Tessel Steering Committee members would leave comments asking questions about our code, which would force us to research more on the issue.
```
Comment (Frijol) [inline Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: :) <3

File: Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+On our third meeting with Kelsey, she had said that "It is better to over-communicate rather than Under-communicate" and thus, we would ( shamelessly 🙈 ) ping them on slack or on the issues whenever we were stuck and a descriptive solution to our questions would follow. Many a times, Kelsey would explain us our doubts on our call. Despite whatever mistakes we would make, the committee members were always positive and ensured that we were able to rectify our errors and learn from them. Many a times, they've tagged us in issues that they thought we should be aware of and sent us links and resources to escalate our learning.
```
Comment (Frijol) [inline Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: "many a times" should be "many times"

File: Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+It was a great experience for both of us because we were able to pass on our knowledge to an enthusiastic crowd, get rid of our fear of speaking in front of unknown people and gaining in depth knowledge of the topics involoved for which we had to explore every nook and corner of them in order to make our sessions productive.
+
+## What's next?
+
+RGSoC provided us with a whole lot of things that we not even imagined of gaining in this interval of three months. As it was a wonderful journey for both of us, we have thought of recommending this program to more and more people by conducting a few related meetups at our University under our Rails Girls Summer of Code, Delhi Chapter. Apart from this, we had discussed way back with our mentor during the application process that we would love to contribute to Tessel even after RGSoC ends. The months might be over, but the memories are here to stay!
```
Comment (Frijol) [inline Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: "that we not even imagined of gaining" should be "that we didn't even imagine gaining"

File: Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+Who would know that two confused, scared and shy humans who had absolutely no idea how they would survive the tech world could successfully complete RGSoC, loaded with experience and confidence?
```
Comment (Frijol) [inline Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: I'm so glad! Thanks for giving me the opportunity to work with you, I really got a lot out of it too.

**Review by Frijol — CHANGES_REQUESTED** [review]

File: Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md (hunk 9 above)
Comment (mittalshravika) [reply Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: Thanks a lot for the reviews. Would be updating the changes soon.

File: Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md (hunk 1 above)
Comment (mittalshravika) [reply Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: Resolved

File: Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md (hunk 4 above)
Comment (mittalshravika) [reply Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: Resolved

Comment (mittalshravika) [reply Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: Resolved

File: Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md (hunk 5 above)
Comment (mittalshravika) [reply Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: Resolved

File: Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md (hunk 7 above)
Comment (mittalshravika) [reply Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: Resolved

File: Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md (hunk 8 above)
Comment (mittalshravika) [reply Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: Resolved

File: Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md (hunk 3 above)
Comment (mittalshravika) [reply Rails-Girls-Summer-Of-Code-2017-Blog-Post-2.md]: Resolved

**Comment by Frijol** [comment]
oo just noticed this! Will try to take a look in the next couple of days

**Comment by Frijol** [comment]
Looking good! I made a copy edit branch pull request – please  have a look: https://github.com/276linesofCode/this-week-in-tessel/pull/1

One further suggestion: it would be great to drop in links to some things you've worked on that you're particularly proud of, e.g. a tutorial you wrote.

RGSOC application deadline (for projects) is coming up on the 25th! I'm hoping we can give the program a boost by publishing this ASAP (though I realize I haven't left a lot of time for further edits). Do you think that's possible?

Thanks!
Kelsey

**Comment by mittalshravika** [comment]
Hey @Frijol, I have merged the pull request that you had made for the grammatical changes.

**Review by Frijol — APPROVED** [review]

**Comment by Frijol** [comment]
Published! http://tesselproject.tumblr.com/post/170042752022/team-276linesofcode-rails-girls-summer-of-code

**Comment by mittalshravika** [comment]
Thanks a lot!!
